File: 1399-count-largest-group/1399-count-largest-group.cpp
class Solution {
public:
    int countLargestGroup(int n) {
        std::map<int, int> group_map;

        for (int i = 1; i <= n; i++) {
            group_map[count_sum(i)]++;
        }

        int m = std::max_element(group_map.begin(), group_map.end(), [](auto a, auto b) { return a.second < b.second; })->second;

        int c = 0;
        for (auto [sum, count] : group_map) 
            if (count == m) c++;
        return c;
    }

    int count_sum(int num) {
        int sum = 0;
        while (num)  {
            sum += num % 10;
            num /=10;
        }
        return sum;
    }
};

File: 1399-count-largest-group/1399-count-largest-group.py
class Solution:
    def count_sum(self, n: int) -> int:
        return sum(int(digit) for digit in str(n))

    def countLargestGroup(self, n: int) -> int:
        group_counts = defaultdict(int)
        for i in range(1, n + 1):
            group_counts[self.count_sum(i)] += 1
        max_count = max(group_counts.values())
        return sum(c == max_count for c in group_counts.values())
